tools/interactive_marker_teleop/src/interactive_marker_teleop.cpp

```diff
@@ -0,0 +1,299 @@
+      switch (feedback->menu_entry_id) {
+        case 1:  // snap to astrobee
+        {
+          // get astrobee position
+          geometry_msgs::TransformStamped t;
+          t = tfBuffer.lookupTransform("world", "body", ros::Time(0));
+          // set marker pose
+          geometry_msgs::Pose p;
+          p.position.x = t.transform.translation.x;  // Help! is there a better way to do this?
+          p.position.y = t.transform.translation.y;
+          p.position.z = t.transform.translation.z;
+          p.orientation = t.transform.rotation;
+          server->setPose("astrobee_teleop", p);
+          server->applyChanges();
+          break;
+        }
+        case 2:  // move to desired position
+        {
+          sendMobilityCommand(ff_msgs::CommandConstants::CMD_NAME_SIMPLE_MOVE6DOF, feedback->pose);
+          break;
+        }
+        case 3:  // dock
+        {
+          sendMobilityCommand(ff_msgs::CommandConstants::CMD_NAME_DOCK);
+          break;
+        }
+        case 4:  // undock
+        {
+          sendMobilityCommand(ff_msgs::CommandConstants::CMD_NAME_UNDOCK);
+          break;
+        }
+        case 5:  // stop
+        {
+          sendMobilityCommand(ff_msgs::CommandConstants::CMD_NAME_STOP_ALL_MOTION);
+          break;
+        }
+        default:
+          break;
+      }
+      break;
+
+      // TODO(jdekarske) check for keep out zones here
+      // case visualization_msgs::InteractiveMarkerFeedback::POSE_UPDATE:
+      //   ROS_INFO_STREAM(s.str() << ": pose changed"
+      //                           << "\nposition = "
+      //                           << feedback->pose.position.x
+      //                           << ", " << feedback->pose.position.y
+      //                           << ", " << feedback->pose.position.z
+      //                           << "\norientation = "
+      //                           << feedback->pose.orientation.w
+      //                           << ", " << feedback->pose.orientation.x
+      //                           << ", " << feedback->pose.orientation.y
+      //                           << ", " << feedback->pose.orientation.z
+      //                           << "\nframe: " << feedback->header.frame_id
+      //                           << " time: " << feedback->header.stamp.sec << "sec, "
+      //                           << feedback->header.stamp.nsec << " nsec");
+      //   break;
+  }
+}
+
+void make6DofMarker(unsigned int interaction_mode, const geometry_msgs::Pose& position) {
+  InteractiveMarker int_marker;
+  int_marker.header.frame_id = "world";
+  int_marker.pose = position;
+  int_marker.scale = 1;
+
+  int_marker.name = "astrobee_teleop";
+  int_marker.description = "Astrobee Command";
+
+  // insert a box
+  makeBoxControl(int_marker);
+  int_marker.controls[0].interaction_mode = interaction_mode;
+
+  InteractiveMarkerControl control;
+
+  tf2::Quaternion orien(1.0, 0.0, 0.0, 1.0);
```

Comment:
Try to avoid uncommon abbreviations, "orientation" would be fine here

Reply:
done